### PR TITLE
decrufting for #378

### PR DESCRIPTION
**Pull request by moldover**

### CODE REVIEW

File: src/main/java/network/brightspots/rcv/ContestConfig.java [hunk 1]
```diff
@@ -393,9 +370,7 @@ private boolean stringAlreadyInUseElsewhere(String string, String field) {
     return inUse;
   }
 
-  // function: candidateStringAlreadyInUseElsewhere
-  // purpose: Takes a candidate name or code and checks for conflicts with other name/codes or other
-  //   strings that are already being used in some other way.
+  // checks for conflicts between a candidate name and other name/codes or other reserved strings
   // param: candidateString is a candidate name or code
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/ContestConfig.java]: did you intend to leave these param comments?

File: src/main/java/network/brightspots/rcv/GuiConfigController.java [hunk 2]
```diff
@@ -859,9 +859,7 @@ protected Void call() {
     // path to config file we will use for tabulation
     private final String configPath;
 
-    // function: TabulatorService
-    // purpose: constructor for Service object which runs a tabulation
-    // param: configPath path to config file to be tabulated
+    // constructor for Service object which runs a tabulation in the background
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/GuiConfigController.java]: I'd rather put this comment before the class declaration, not on the constructor

File: src/main/java/network/brightspots/rcv/StreamingCVRReader.java [hunk 3]
```diff
@@ -113,66 +109,49 @@
   private static Pair<Integer, Integer> getCellAddress(String address) {
     // this regex will parse a string into
     // a sequence of one or more non-digits followed by a sequence of one or more digits
-    // and store these substrings into addressParts array
     String[] addressParts = address.split("(?<=\\D)(?=\\d)");
     if (addressParts.length != 2) {
       Logger.log(Level.SEVERE, "Invalid cell address: %s", address);
       throw new InvalidParameterException();
     }
-    // row is the 0-based row of the cell
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/StreamingCVRReader.java]: this and the subsequent comment are actually kind of helpful

File: src/main/java/network/brightspots/rcv/StreamingCVRReader.java [hunk 4]
```diff
@@ -113,66 +109,49 @@
...
   // purpose: Handle empty cells encountered while parsing a CVR.  Unlike empty rows, empty cells
   // do not trigger parsing callbacks so their existence must be inferred and handled when they
   // occur in a rankings cell.
-  // param: currentRank the rank at which we stop inferring empty cells.
+  // param: currentRank the rank at which we stop inferring empty cells for this invocation
   private void handleEmptyCells(int currentRank) {
-    // rank iterates between lastRankSeen and currentRank adding audit data and UWI rankings
     for (int rank = lastRankSeen + 1; rank < currentRank; rank++) {
-      // add data to audit log
+      // add "empty cell" to audit log
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/StreamingCVRReader.java]: remove

File: src/main/java/network/brightspots/rcv/Tabulator.java [hunk 5]
```diff
@@ -92,49 +85,39 @@
...
+  // run the main tabulation routine to determine contest results
   // returns: set containing winner(s)
   Set<String> tabulate() throws TabulationCancelledException {
     if (config.needsRandomSeed()) {
       Random random = new Random(config.getRandomSeed());
       if (config.getTiebreakMode() == TieBreakMode.GENERATE_PERMUTATION) {
-        // sort candidate permutation for reproducibility
+        // sort candidate permutation first for reproducibility
         Collections.sort(config.getCandidatePermutation());
+        // every day I'm shuffling
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/Tabulator.java]: are you, though?

File: src/main/java/network/brightspots/rcv/Tabulator.java [hunk 6]
```diff
@@ -211,16 +192,13 @@
         }
       } else if (winnerToRound.size() < config.getNumberOfWinners()
           || (config.isSingleSeatContinueUntilTwoCandidatesRemainEnabled()
-              && candidateToRoundEliminated.size() < config.getNumCandidates() - 2)) {
+          && candidateToRoundEliminated.size() < config.getNumCandidates() - 2)) {
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/Tabulator.java]: this indentation seems wrong

File: src/main/java/network/brightspots/rcv/Tabulator.java [hunk 7]
```diff
@@ -947,17 +877,15 @@ private void recordSelectionForCastVoteRecord(
...
           for (String candidate : candidateSet) {
             if (candidatesSeen.contains(candidate)) {
-              // duplicate is found, exit the loop so we can exhaust the cvr
               duplicateCandidate = candidate;
               break;
             }
             candidatesSeen.add(candidate);
           }
-          // if duplicate was found, exhaust cvr
+          // if duplicate was found exhaust cvr
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/Tabulator.java]: aggressive

File: src/main/java/network/brightspots/rcv/Tabulator.java [hunk 8]
```diff
@@ -1206,18 +1111,14 @@ public String toString() {
...
-    // returns: the new object
     BatchElimination(String candidateId, BigDecimal runningTotal, BigDecimal nextHighestTally) {
       this.candidateId = candidateId;
       this.runningTotal = runningTotal;
       this.nextHighestTally = nextHighestTally;
     }
   }
 
-  static class TabulationCancelledException extends Exception {}
+  static class TabulationCancelledException extends Exception {
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/Tabulator.java]: why?

File: src/main/java/network/brightspots/rcv/TieBreak.java [hunk 9]
```diff
@@ -51,35 +51,27 @@
...
-  // candidate ID selected to lose/win the tiebreak
   private String selectedCandidate;
-  // reason for the selection
   private String explanation;
 
-  // function: TieBreak
-  // purpose: TieBreak constructor stores member data
-  // param: selectingAWinner whether we're determining a winner (as opposed to a loser)
+  // TieBreak constructor
+  // param: selectingAWinner are we determining a winner or loser
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/TieBreak.java]: just curious how you decide when to list out the params like this vs. removing them

File: src/main/java/network/brightspots/rcv/ContestConfig.java (hunk 1 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/ContestConfig.java]: yes.  the meaning of 'field' is not obvious

File: src/main/java/network/brightspots/rcv/Tabulator.java (hunk 5 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/Tabulator.java]: you're not always around to see it, but yes

File: src/main/java/network/brightspots/rcv/Tabulator.java (hunk 6 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/Tabulator.java]: that's what "intelli" J gives me

File: src/main/java/network/brightspots/rcv/Tabulator.java (hunk 7 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/Tabulator.java]: you think it's a game?

File: src/main/java/network/brightspots/rcv/Tabulator.java (hunk 8 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/Tabulator.java]: again, it's what intelliJ -- presumably form the google code style jar does

File: src/main/java/network/brightspots/rcv/TieBreak.java (hunk 9 above)
Comment (moldover) [reply src/main/java/network/brightspots/rcv/TieBreak.java]: Generally I kept param comments when the meaning was not obvious, and removed them when it was.  In this case selectingAWinner being false does not naturally imply selecting a loser so the comment is important.  I confirmed this with Eric.

**Pull request merged**